dali/pipeline/executor/queue_policy.h

```diff
@@ -138,26 +138,38 @@ struct UniformQueuePolicy {
     }
   }
 
-  void SignalError() {
-    exec_error_ = true;
+  void NotifyAll() {
     ready_cond_.notify_all();
     free_cond_.notify_all();
   }
 
-  bool IsErrorSignaled() {
-    return exec_error_;
+  void SignalStop() {
+    {
+      std::unique_lock<std::mutex> lock(ready_mutex_);
```

Comment:
lock_guard would be more appropiate here

Reply:
```suggestion
      std::lock_guard<std::mutex> lock(ready_mutex_);
```